race condition in naml stuffs

./naml -f sample/sample.naml list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x46e8a2]

goroutine 1 [running]:
github.com/kris-nova/naml.AddRPC(0xc0001d91a0, 0x12, 0xc0004f1c90, 0x1)
	/home/dcooley/git/naml/rpc.go:303 +0x514
github.com/kris-nova/naml.AllInit(0x18fd24b, 0xe, 0x0, 0xc0004f1b10, 0x1, 0x1, 0xc0004f1c00, 0xc0004f1c20)
	/home/dcooley/git/naml/cmd.go:353 +0x1fd
github.com/kris-nova/naml.RunCommandLineWithOptions.func5(0xc000558040, 0x1, 0x1)
	/home/dcooley/git/naml/cmd.go:285 +0x65
github.com/urfave/cli/v2.(*Command).Run(0xc00040d9e0, 0xc00047bf80, 0x0, 0x0)
	/home/dcooley/git/naml/vendor/github.com/urfave/cli/v2/command.go:163 +0x4dd
github.com/urfave/cli/v2.(*App).RunContext(0xc000261ba0, 0x1af9dd0, 0xc0001b4320, 0xc0001bc000, 0x4, 0x4, 0x0, 0x0)
	/home/dcooley/git/naml/vendor/github.com/urfave/cli/v2/app.go:313 +0x810
github.com/urfave/cli/v2.(*App).Run(...)
	/home/dcooley/git/naml/vendor/github.com/urfave/cli/v2/app.go:224
github.com/kris-nova/naml.RunCommandLineWithOptions(0x408005, 0xc000182058)
	/home/dcooley/git/naml/cmd.go:316 +0xf2b
github.com/kris-nova/naml.RunCommandLine(...)
	/home/dcooley/git/naml/cmd.go:51
github.com/kris-nova/naml.RunCommandLineAndExit()
	/home/dcooley/git/naml/cmd.go:39 +0x25
main.main()
	/home/dcooley/git/naml/cmd/main.go:30 +0x25

--- a/cmd.go
+++ b/cmd.go
@@ -350,10 +350,16 @@ func AllInit(kubeConfigPath string, verbose bool, with []string) error {
 	// [ Child Runtime System ]
 	if len(with) > 0 {
 		for _, childPath := range with {
-			err := AddRPC(childPath)
-			if err != nil {
-				logger.Warning("Unable to add child naml %s: %v", childPath, err)
+			for i := 0; i < 3; i++ {
+				err := AddRPC(childPath)
+				if err != nil {
+					logger.Warning("Unable to add child naml %s: %v", childPath, err)
+					time.Sleep(time.Millisecond * 20)
+				} else {
+					break
+				}
 			}
+
 		}
 	}
 

--- a/rpc.go
+++ b/rpc.go
@@ -263,9 +263,7 @@ func AddRPC(path string) error {
 
 	logger.Info("Starting IPC with remote RPC server %s", file.Name())
 
-	// Execute the child binary and pass in "c" for
-	// shorthand to tell the child to run in child runtime mode.
-	// logger.Info("Command %s c", path)
+	// Execute another naml binary and pass in "rpc" to run in rpc child mode
 	cmd := exec.Command(absolutePath, "rpc")
 	childOut := &bytes.Buffer{}
 	childErr := &bytes.Buffer{}
@@ -297,8 +295,11 @@ func AddRPC(path string) error {
 			break
 		default:
 			switch {
-			case childOut.Len() > 1:
+			case childOut.Len() > 2:
 				message := childOut.Bytes()
+				if message == nil {
+					continue
+				}
 				// According to the naml RPC this should be the TCP port
 				rpcHello := string(message)
 				rpcSplit := strings.Split(rpcHello, "\n")

